app/[...slug]/page.tsx

```diff
@@ -6,22 +6,34 @@ import { getPageMetadata } from '@/utilities/getPageMetadata';
 
 const activeEnv = process.env.NODE_ENV || 'development';
 
-export async function generateMetadata({ params }): Promise<Metadata> {
-  const { data } = await getStoryData(params);
-  const blok = data.story.content;
+type PathsType = {
+  slug: string[];
+};
 
-  let slug: string = params.slug ? params.slug.join('/') : '';
-  const meta = getPageMetadata({ blok, slug });
+type ParamsType = {
+  slug: string[];
+};
 
-  return meta;
-}
+async function generateStaticParams() {
+  const storyblokApi: StoryblokClient = getStoryblokApi();
+  let sbParams: ISbStoriesParams = {
+    version: activeEnv === 'development' ? 'draft' : 'published',
+    cv: activeEnv === 'development' ? Date.now() : undefined,
+  };
 
-export default async function Page({ params }) {
-  const { data } = await getStoryData(params);
+  const { data: { links } } = await storyblokApi?.get('cdn/links', sbParams);
+  let paths: PathsType[] = [];
```

Comment:
In this file page.tsx, I'm mostly rearranging things in a more logical way (doesn't affect outcome but more logical for anyone who will be reading the code, e.g., we find the list of paths/slugs first before we fetch the data for that path). Also typed a few params here.